include/ginkgo/core/matrix/dense.hpp

```diff
@@ -786,12 +784,19 @@ class Dense
      *                (the number of columns in the vector must match the number
      *                of columns of this)
      */
-    void compute_norm2(LinOp* result) const
-    {
-        auto exec = this->get_executor();
-        this->compute_norm2_impl(
-            make_temporary_output_clone(exec, result).get());
-    }
+    void compute_norm2(LinOp* result) const;
```

Comment:
this is not virtual, so you can make the tmp = {} with default?

Reply:
Keeping them separate makes it easier to write separate documentation for them. Also I'm not sure you can have a non-const lvalue reference parameter with default value anyways.